addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/AbstractMultipleOptionsTablePanel.java

```diff
@@ -0,0 +1,119 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.selenium;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.JButton;
+import javax.swing.event.TableModelEvent;
+import javax.swing.event.TableModelListener;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.utils.EnableableInterface;
+import org.zaproxy.zap.view.AbstractMultipleOptionsBaseTablePanel;
+import org.zaproxy.zap.view.AbstractMultipleOptionsTableModel;
+
+/**
+ * This is a copy of the core class with the same name, changed to expose the constructor with
+ * allowModification. This class should be replaced with the core class once a version of that is
+ * available with that constructor visible.
+ */
+public abstract class AbstractMultipleOptionsTablePanel<E extends EnableableInterface>
+        extends AbstractMultipleOptionsBaseTablePanel<E> {
+
+    private static final long serialVersionUID = 1L;
+    private static final String ENABLE_ALL_BUTTON_LABEL =
+            Constant.messages.getString("multiple.options.panel.enableAll.button.label");
+    private static final String DISABLE_ALL_BUTTON_LABEL =
+            Constant.messages.getString("multiple.options.panel.disableAll.button.label");
+
+    private JButton enableAllButton;
+    private JButton disableAllButton;
+
+    public AbstractMultipleOptionsTablePanel(AbstractMultipleOptionsTableModel<E> model) {
+        this(model, true);
+    }
+
+    public AbstractMultipleOptionsTablePanel(
+            AbstractMultipleOptionsTableModel<E> model, boolean allowModification) {
+        super(model, allowModification);
+
+        final boolean buttonsEnabled = getModel().getRowCount() > 0;
+
+        enableAllButton = new JButton(ENABLE_ALL_BUTTON_LABEL);
+        enableAllButton.setEnabled(buttonsEnabled);
+        enableAllButton.addActionListener(
+                new ActionListener() {
+
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        ((AbstractMultipleOptionsTableModel<?>) getMultipleOptionsModel())
+                                .setAllEnabled(true);
+                    }
+                });
+
+        disableAllButton = new JButton(DISABLE_ALL_BUTTON_LABEL);
+        disableAllButton.setEnabled(buttonsEnabled);
+        disableAllButton.addActionListener(
+                new ActionListener() {
+
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        ((AbstractMultipleOptionsTableModel<?>) getMultipleOptionsModel())
+                                .setAllEnabled(false);
+                    }
+                });
```

Comment:
Could be simplified/shortened with lambdas?

Reply:
This is a copy of a core class (which will be replaced), not worth changing IMO.

Reply:
Okay